internal/http/services/owncloud/ocs/conversions/main.go

```diff
@@ -282,25 +281,20 @@ func PublicShare2ShareData(share *link.PublicShare, r *http.Request) *ShareData
 	}
 
 	return &ShareData{
-		// THERE BE DRAGONS
-		// TODO map share.resourceId to path and storage ... requires a stat call
 		// share.permissions ar mapped below
-		// TODO lookup user metadata
 		// DisplaynameOwner:     creator.DisplayName,
-		// TODO lookup user metadata
 		// DisplaynameFileOwner: share.GetCreator().String(),
 		ID:           share.Id.OpaqueId,
-		Permissions:  publicSharePermissions2OCSPermissions(share.GetPermissions()),
 		ShareType:    ShareTypePublicLink,
-		UIDOwner:     UserIDToString(share.Creator),
 		STime:        share.Ctime.Seconds, // TODO CS3 api birth time = btime
-		UIDFileOwner: UserIDToString(share.Owner),
 		Token:        share.Token,
 		Expiration:   expiration,
 		MimeType:     share.Mtime.String(),
 		Name:         r.FormValue("name"),
-		// URL:                  r.Host + "/#/s/" + share.Token, // this is broken. r.Host doesn't point to Phoenix
-		URL: "localhost:8300/#/s/" + share.Token,
+		URL:          r.Header.Get("Origin") + "/#/s/" + share.Token,
```

Comment:
Shouldn't Phoenix use this path as relative? Having the hostname here means OCIS server needs to know where Phoenix is running

Reply:
indeed it should, but as far as my knowledge goes it doesn't. Correct me if I'm wrong @butonic

Reply:
cc'ing @kulmann perhaps you can also shed some light as per why phoenix is not dealing with relative paths there. I'd give it another try in my local environment but I'm pretty sure it will `burp`

Reply:
someone needs to construct the url. we can do it in the clients or on the server side. We could return a relative path as the URL , but be aware that it might require special handling if the first path segment contains a `:`, see https://tools.ietf.org/html/rfc3986#section-4.2

IMO we should teach the clients to handle relative paths correctly. Including desktop and mobile.

Reply:
Merging from now, but I'm creating an issue to keep track of it